### PR TITLE
Implement CLI query subcommands (read/range)

### PR DESCRIPTION
**Pull request by MaxJTRice**
Here is the test document: https://docs.google.com/document/d/12wJ5ULLgLG9rgEGFF-Y3hOj1briu_jQF6Bmm0g5gVhg/edit#

### CODE REVIEW

File: names.go [hunk 1]
```diff
@@ -35,7 +35,8 @@ var (
 	namePrefixRegex = regexp.MustCompile("^[a-z_][a-z0-9_.]{0,31}$")
 )
 
-func isValidNamePrefix(namePrefix string) error {
+// IsValidNamePrefix checks if a prefix name conforms the regular expression
```
[the remark this answers is not in the document]
Comment (MaxJTRice) [reply names.go]: agree!

File: cmd/dosa/client.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+			fieldName, ok := colToField[columnName]
+			if !ok {
+				continue // we ignore fields that we don't know about
+			}
+			rowField[fieldName] = fieldValue
+		}
+		rowsField[idx] = rowField
+	}
+	return rowsField
+}
```
[the remark this answers is not in the document]
Comment (MaxJTRice) [reply cmd/dosa/client.go]: In this file we define the new client used by query subcommand with implementation its API. Read and Range are all APIs of the client and it makes more sense to me to keep them together

File: cmd/dosa/client.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
```
[the remark this answers is not in the document]
Comment (MaxJTRice) [reply cmd/dosa/client.go]: :-)
Do I also need to change this for all other files I don't touch?

File: cmd/dosa/query.go [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+type QueryCmd struct {
+	*QueryOptions
+	Scope         scopeFlag `short:"s" long:"scope" description:"Storage scope for the given operation." required:"true"`
+	NamePrefix    string    `short:"n" long:"namePrefix" description:"Name prefix for schema types."`
+	Prefix        string    `short:"p" long:"prefix" description:"Name prefix for schema types." hidden:"true"`
+	Path          string    `long:"path" description:"path to source." required:"true"`
+	provideClient queryClientProvider
+}
+
+func (c *QueryCmd) getNamePrefix() (string, error) {
```
[the remark this answers is not in the document]
Comment (MaxJTRice) [reply cmd/dosa/query.go]: make sense!

File: cmd/dosa/query.go [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+
+func printResult(res []map[string]dosa.FieldValue) {
+	fmt.Println("Results:")
+	for _, fvs := range res {
+		for k, v := range fvs {
+			fmt.Printf("%s: %v\n", k, reflect.Indirect(reflect.ValueOf(v)))
+		}
+		fmt.Printf("\n")
+	}
+}
```
[the remark this answers is not in the document]
Comment (MaxJTRice) [reply cmd/dosa/query.go]: Actually 'read' and 'range' are all subcommands of 'query' subcommand. The usage is like:
dosa query read ...
and
dosa query range ...
I'd rather keep them together like subcommands of 'schema' and 'scope'

File: cmd/dosa/client.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (phliar) [reply cmd/dosa/client.go]: Why does it make more sense? Large files are more awkward to browse. Go allows an object's methods to be spread across files, we should use that to increase readability.

[the remark this answers is not in the document]
Comment (MaxJTRice) [reply cmd/dosa/client.go]: Sounds reasonable! Gonna have an update to split the files.

**Pull request merged**